processor/removeemptyvaluesprocessor/processor.go

```diff
@@ -0,0 +1,266 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package removeemptyvaluesprocessor
+
+import (
+	"context"
+	"strings"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+)
+
+type emptyValueProcessor struct {
+	logger               *zap.Logger
+	c                    Config
+	excludeResourceKeys  []MapKey
+	excludeAttributeKeys []MapKey
+	excludeBodyKeys      []MapKey
+}
+
+func newEmptyValueProcessor(logger *zap.Logger, cfg Config) *emptyValueProcessor {
+	var excludeResourceKeys []MapKey
+	var excludeAttributeKeys []MapKey
+	var excludeBodyKeys []MapKey
```

Comment:
Would it be better to store these as a Set basically a `map[string]struct{}` so searching for a match is O(1) rather than O(len(exclude))?